### ISSUE TITLE
Lint/UnneededSplatExpansion can get very confused

### ISSUE THREAD

**Issue by marcandre** [bbatsov/rubocop#5054]
```
foo = [1, 2, *Array.new(how_many_nils), 4, 5]
```

Lint/UnneededSplatExpansion is flagged for this on the `*Array.new`.

Moreover, it gets "fixed" as:

```
foo = [1, 2, [Array.new(how_many_nils)], 4, 5]
```

That code is simply not equivalent!

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.5 x86_64-darwin15)
```
